Helper/Product.php

```diff
@@ -0,0 +1,336 @@
+     * @param CatalogProductHelper $productHelper
+     * @param ProductData $sailthruProduct
+     * @param Settings $sailthruSettings
+     * @param StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        Context $context,
+        ClientManager $clientManager,
+        Configurable $configurableProduct,
+        ImageHelper $imageHelper,
+        CatalogProductHelper $productHelper,
+        SailthruProduct $sailthruProduct,
+        Settings $sailthruSettings,
+        StoreManagerInterface $storeManager
+    ) {
+        $this->clientManager = $clientManager;
+        $this->configurableProduct = $configurableProduct;
+        $this->imageHelper = $imageHelper;
+        $this->productHelper = $productHelper;
+        $this->sailthruProduct = $sailthruProduct;
+        $this->sailthruSettings = $sailthruSettings;
+        $this->storeManager = $storeManager;
+
+        parent::__construct($context);
+    }
+
+    /**
+     * Send post content to Sailthru
+     *
+     * @param ProductModel $product
+     *
+     * @return Product
+     *
+     * @throws NoSuchEntityException
+     */
+    public function postContent(ProductModel $product)
+    {
+        if ($this->isAllStoreViews($product)) {
+            $this->sendMultipleRequests($product, self::EVENT_TYPE_SAVE_PRODUCT);
+        } else {
+            if ($this->sailthruProduct->isProductInterceptOn($product->getStoreId())) {
+                $this->sendRequest($product, self::EVENT_TYPE_SAVE_PRODUCT);
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * Send delete content to Sailthru
+     *
+     * @param ProductModel $product
+     *
+     * @return Product
+     *
+     * @throws NoSuchEntityException
+     */
+    public function deleteContent(ProductModel $product)
+    {
+        if ($this->isAllStoreViews($product)) {
+            $this->sendMultipleRequests($product, self::EVENT_TYPE_DELETE_PRODUCT);
+        } else {
+            if ($this->sailthruProduct->isProductInterceptOn($product->getStoreId())) {
+                $this->sendRequest($product, self::EVENT_TYPE_DELETE_PRODUCT);
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * To send multiple requests to API.
+     *
+     * @param ProductModel $product
+     * @param string $eventType
+     *
+     * @return Product
+     */
+    protected function sendMultipleRequests(ProductModel $product, string $eventType)
+    {
+        $storeIds = $product->getStoreIds();
+        foreach ($storeIds as $storeId) {
+            $this->sendRequest($product, $eventType, (int)$storeId);
+        }
+
+        return $this;
+    }
+
+    /**
+     * To send single request to API.
+     *
+     * @param ProductModel $product
+     * @param int|null $storeId
+     * @param string $eventType
+     *
+     * @return Product
+     */
+    protected function sendRequest(ProductModel $product, string $eventType, int $storeId = null)
```

Comment:
Please refactor entire helper that way so that you do not have dual behavior on all protected methods based on $eventType. Simply create separate methods for each case - when product being added and product being deleted.

Remove dynamic usages like so:
```            $client->_eventType = $eventType;
            $apiMethod = 'api' . self::METHOD_TYPES[$eventType];
            $client->$apiMethod('content', $data);
```

and always make explicit client method calls and explicit `$client->_eventType` setting